app/src/features/apiClient/screens/apiClient/components/views/http/components/HttpClientUrl/HttpClientUrl.tsx

```diff
@@ -15,15 +17,40 @@ interface ApiClientUrlProps {
   currentEnvironmentVariables: ScopedVariables;
   onEnterPress: (e: KeyboardEvent) => void;
   onUrlChange: (value: string, finalParams: KeyValuePair[]) => void;
+  onImportCurlRequest?: (request: RQAPI.Request) => void;
 }
 
-const HttpApiClientUrl = ({ url, currentEnvironmentVariables, onEnterPress, onUrlChange }: ApiClientUrlProps) => {
+//prettier-ignore
+const HttpApiClientUrl = ({ url, currentEnvironmentVariables, onEnterPress, onUrlChange, onImportCurlRequest }: ApiClientUrlProps) => {
   const [queryParams, setQueryParams] = useQueryParamStore((state) => [state.queryParams, state.setQueryParams]);
 
   const updatePathVariableKeys = usePathVariablesStore((state) => state.updateVariableKeys);
 
   const handleUrlChange = useCallback(
     (value: string) => {
+      const trimmedValue = value.trim();
+
+      if (trimmedValue.toLowerCase().startsWith("curl ")) {
+        try {
+          const requestFromCurl = parseCurlRequest(trimmedValue);
+
+          if (!requestFromCurl || !requestFromCurl.url) {
+            toast.error("Failed to parse cURL command. Please ensure it is valid.");
+            return;
+          }
+          onImportCurlRequest(requestFromCurl);
+
+          const urlWithoutQueryParams = requestFromCurl.url.split("?")[0];
+          const finalQueryParams: KeyValuePair[] = (requestFromCurl as any).queryParams || [];
+
+          onUrlChange(urlWithoutQueryParams, finalQueryParams);
```

Comment:
This can be done like this too instead of parsing and creating entry object by ourself

```
    const requestFromCurl = parseCurlRequest(trimmedValue);
    const apiEntry = getEmptyApiEntry(RQAPI.ApiEntryType.HTTP, requestFromCurl);
    setEntry(apiEntry); // This needs to be passed from parent?

```